### DISCUSSION TITLE
Consequence of adding another master to a 3-node master.

### DISCUSSION THREAD

**Discussion by parseltongued** [okd-project/okd#1931] [General]
Hi, 

I've a 3-master node OKD setup using an external HAProxy LB. 

I'm installing min.io and the minimum requirement is 4 nodes. I'm thinking of adding a new master to the cluster to meet the minimum requirement.

Whilst I understand a HA cluster requires 3 nodes to achieve quorum, what's the consequence OKD will have on having a 4 node cluster?

Thank you.

**Comment by msteenhu** [comment]
This is a Kubernetes/etcd question, or even Raft (the consensus algorithm) question. Explained nicely here for example: https://thenewstack.io/how-many-nodes-for-your-kubernetes-control-plane/


> etcd will always demand that more than half of the nodes be alive and reachable in order to have a quorum, so the database can perform updates.
>   
>   Thus a 2-node control plane requires both nodes to be up. A 3-node control plane also requires that 2 nodes be up. A 4-node control plane requires 3 nodes to be up. So a 4-node control plane has worse availability than a 3-node control plane — as while both can suffer a single node outage, and neither can deal with a 2-node outage, the odds of that happening in a 4-node cluster are higher
> 

Normally your load runs on worker nodes. If you use your masters as workers too, add 1 worker only node? Or at least make sure etcd is not provisioned on there.

**Reply by parseltongued** [reply]
thanks for the insightful article and suggestion.
